### PR TITLE
caps word implementation

### PR DESCRIPTION
**Pull request by masiullah**
This is a similar feature available in some of the forks of qmk allowing to break out of caps lock if you type keys that break the word
Please review and provide feedback if this is in the correct direction
This is my first contribution as I am learning Python and experimenting with the kb2040 I got recently

### CODE REVIEW

**Comment by kdb424** [comment]
I like this functionality, and want to hear what one of our reviewers says about the code, but from a glance, I think this looks fairly good!

File: kmk/modules/capsword.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        # enables/disables capsword
+        if key == KC.CW and is_pressed:
+            if not self._cw_active:
+                self._cw_active = True
+                self.discard_timeout(keyboard)
+                self.request_timeout(keyboard)
+            else:
+                self.discard_timeout(keyboard)
+                self.process_timeout()
```
Comment (xs5871) [inline kmk/modules/capsword.py:65]: This should go in the `KC.CW.on_press` handler.

File: kmk/modules/capsword.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            continue_cw = False
+            # capitalize alphabets
+            if key.code in self.alphabets:
+                continue_cw = True
+                keyboard.process_key(KC.LSFT, is_pressed)
+            elif (
+                key.code in self.numbers
+                or isinstance(key, ModifierKey)
+                or key in self.keys_ignored
+                or key.code >= 1000  # user defined keys are also ignored
```
Comment (xs5871) [inline kmk/modules/capsword.py:46]:
```suggestion
                or key.code >= FIRST_KMK_INTERNAL_KEY  # user defined keys are also ignored
```
for consistency: let's reference magic numbers by name.

File: kmk/modules/capsword.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                'CAPSWORD',
+                'CW',
+            ),
+        )
+
+    def during_bootup(self, keyboard):
+        return
+
+    def matrix_detected_press(self, keyboard):
+        return keyboard.matrix_update is None
```
Comment (xs5871) [inline kmk/modules/capsword.py:30]: This method is never used.

**Review by xs5871 — COMMENTED** [review]
I've got some comments, nothing major.
Docs would be nice at some point, of course.
I'm almost certain I could break it with the right unit-test, but that's a bridge we can cross when there are complaints, to be honest.

File: kmk/modules/capsword.py (hunk 2 above)
Comment (masiullah) [reply kmk/modules/capsword.py:46]: Sure, will definitely

File: kmk/modules/capsword.py (hunk 3 above)
Comment (masiullah) [reply kmk/modules/capsword.py:30]: I will remove this method

File: kmk/modules/capsword.py (hunk 1 above)
Comment (masiullah) [reply kmk/modules/capsword.py:65]: I will move this to the handler

**Comment by masiullah** [comment]
> I've got some comments, nothing major. Docs would be nice at some point, of course. I'm almost certain I could break it with the right unit-test, but that's a bridge we can cross when there are complaints, to be honest.

I would also want to do unit test cases from the computer. Is there any doc or guidance available that can help me perform those UT cases

**Comment by xs5871** [comment]
There's pretty much no documentation, and be warned, the unit test code is very bare bones.
Tests are under `tests/`. You can take a look at the `test_*.py` files for reference and run them with `make unit-tests`.
I wouldn't mind approving the PR without a unit test for now. You can always submit it later.

**Comment by masiullah** [comment]
Please review and let me know your thoughts

**Review by xs5871 — APPROVED** [review]

**Review by kdb424 — APPROVED** [review]

**Comment by kdb424** [comment]
Merging based on a passed review.

**Pull request merged**